zerver/lib/notifications.py

```diff
@@ -298,10 +299,10 @@ def handle_missedmessage_emails(user_profile_id, missed_email_events):
     if not receives_offline_notifications(user_profile):
         return
 
-    messages = [um.message for um in UserMessage.objects.filter(user_profile=user_profile,
-                                                                message__id__in=message_ids,
-                                                                flags=~UserMessage.flags.read)]
-    if not messages:
+    messages = Message.objects.filter(usermessage__user_profile_id=user_profile,
+                                      id__in=message_ids,
+                                      usermessage__flags=~UserMessage.flags.read)
+    if not messages.exists():
         return
```

Comment:
On the contrary, I've decrease queries quantity in this place. but yes it would be better to use boolean condition here for queryset.